### PR TITLE
Update LSUN Dataset class

### PR DESCRIPTION
**Pull request by Jaesuny**
- Fix uninitialized instance variable in `LSUN.__repr__`

- Maintain consistency with other Dataset classes

### CODE REVIEW

File: torchvision/datasets/lsun.py [hunk 1]
```diff
@@ -69,13 +70,17 @@ class LSUN(data.Dataset):
...
                  transform=None, target_transform=None):
         categories = ['bedroom', 'bridge', 'church_outdoor', 'classroom',
                       'conference_room', 'dining_room', 'kitchen',
                       'living_room', 'restaurant', 'tower']
         dset_opts = ['train', 'val', 'test']
-        self.db_path = db_path
+        self.root = os.path.expanduser(root)
+        self.transform = transform
+        self.target_transform = target_transform
+        self.classes = classes
```
Comment (soumith) [inline torchvision/datasets/lsun.py]: self.classes = classes is already present in line 104

Comment (Jaesuny) [reply torchvision/datasets/lsun.py]: You're right.
I fixed this.

File: torchvision/datasets/lsun.py [hunk 2]
```diff
@@ -96,13 +101,12 @@ def __init__(self, db_path, classes='train',
                                      'Options are: ' + str(dset_opts)))
         else:
             raise(ValueError('Unknown option for classes'))
-        self.classes = classes
```
Comment (fmassa) [inline torchvision/datasets/lsun.py]: This doesn't look right: there is a few operations that are performed just before that modify `classes`, and you don't take that into account when moving `self.classes` up.

File: torchvision/datasets/lsun.py [hunk 3]
```diff
@@ -69,13 +70,17 @@ class LSUN(data.Dataset):
...
-    def __init__(self, db_path, classes='train',
+    def __init__(self, root, classes='train',
                  transform=None, target_transform=None):
         categories = ['bedroom', 'bridge', 'church_outdoor', 'classroom',
                       'conference_room', 'dining_room', 'kitchen',
                       'living_room', 'restaurant', 'tower']
         dset_opts = ['train', 'val', 'test']
-        self.db_path = db_path
+        self.root = os.path.expanduser(root)
+        self.transform = transform
```
Comment (fmassa) [inline torchvision/datasets/lsun.py]: `transform` is not used in `self`, but it is passed to `LSUNClass` so this is unnecessary.

Comment (fmassa) [reply torchvision/datasets/lsun.py]: oh, looks like this is for `__repr__`. Sounds good then

**Comment by soumith** [comment]
thanks @Jaesuny !

**Comment by chenguanfu511** [comment]
How can I create the LSUNClass LMDB?

**Comment by fmassa** [comment]
@chenguanfu511 I don't know, I have never used LSUN, but I would expect that the original datasets are in LMDB format?

**Pull request merged**